File: Src/Icm.Core/TimerExtensions.vb
Imports System.Runtime.CompilerServices

Namespace Icm.Timers

    Public Module TimerExtensions

        <Extension>
        Function GetNextElapsed(timer As System.Timers.Timer) As Date
            Return Now.AddMilliseconds(timer.Interval)
        End Function

        <Extension>
        Sub SetNextElapsed(timer As System.Timers.Timer, d As Date)
            timer.Interval = d.Subtract(Now).TotalMilliseconds
        End Sub

    End Module

End Namespace

File: Src/Icm.Core/TimerExtensions.cs
using System;

namespace Icm.Timers
{
	public static class TimerExtensions
	{
		public static DateTime GetNextElapsed(this System.Timers.Timer timer)
		{
			return DateTime.Now.AddMilliseconds(timer.Interval);
		}

		public static void SetNextElapsed(this System.Timers.Timer timer, DateTime d)
		{
			timer.Interval = d.Subtract(DateTime.Now).TotalMilliseconds;
		}
	}
}